### PR TITLE
New Hook Key API Management

### PR DESCRIPTION
**Pull request by firoozehmehri-okta**
## Description:
Add new Key Management section 
- **Is this PR related to a Monolith release?** 

### Resolves:

* [OKTA-524831]https://oktainc.atlassian.net/browse/OKTA-524831

[OKTA-524831] : https://oktainc.atlassian.net/browse/OKTA-524831?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[OKTA-524831]: https://oktainc.atlassian.net/browse/OKTA-524831?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[OKTA-524831]: https://oktainc.atlassian.net/browse/OKTA-524831?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by alexshvidky-okta** [comment]
Looks good to me

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+title: Key Management
+category: management
+excerpt:
+  The Key Management API provides a CRUD interface for registering key
+  endpoints to use with Hooks.
+---
+
+# Key Management API
+
+The Okta key management API provides a CRUD interface for JSON Web Keys (JWK) used with other parts of the application, such as inline hooks. For information on how to create inline hooks, see [Inline hooks management API](/docs/reference/api/inline-hooks/).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:11]:
```suggestion
The Okta Key Management API provides a CRUD interface for JSON Web Keys (JWK) used with other parts of the application, such as inline hooks. For information on how to create inline hooks, see [Inline hooks management API](/docs/reference/api/inline-hooks/).
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+
+## Get started
+
+Explore the key management API:[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/a36f6c129009072f78b5?action=collection%2Fimport)
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:15]:
```suggestion
Explore the Key Management API:[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/a36f6c129009072f78b5?action=collection%2Fimport)
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+
+## Key management operations
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:17]:
```suggestion
## Key Management operations
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+
+The key management API has the following CRUD operations:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:19]:
```suggestion
The Key Management API has the following CRUD operations:
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+
+> **Note:** The new key that you create is accessed by the key name for use with inline hooks.
+
+The total number of keys that you can create in an Okta org is limited to 50.
+
+##### Request parameters
+
+| Parameter   | Description                                                                                  | Param Type   | DataType                                    | Required |
+| ----------- | -------------------------------------------------------------------------------------------- | ------------ | ------------------------------------------- | -------- |
+| key | A valid request object that specifies the details of the key that you are creating   | Body         | [key request object](#key-request-object)  | TRUE     |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:42]:
```suggestion
| key | A valid request object that specifies the details of the key that you create   | Body         | [key request object](#key-request-object)  | TRUE     |
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+* [Get a key](#get-a-key)
+* [List keys](#list-keys)
+* [Get a public key](#get-a-public-key)
+* [Update a key](#update-a-key)
+* [Delete a key](#delete-a-key)
+
+### Create a key
+
+<ApiOperation method="post" url="/api/v1/hook-keys" />
+
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:31]: Creates a key for use with inline hooks

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+##### Request parameters
+
+| Parameter   | Description                                                                                  | Param Type   | DataType                                    | Required |
+| ----------- | -------------------------------------------------------------------------------------------- | ------------ | ------------------------------------------- | -------- |
+| key | A valid request object that specifies the details of the key that you are creating   | Body         | [key request object](#key-request-object)  | TRUE     |
+
+##### Response parameters
+
+The response is a [key object](#key-object) that represents the key that was created. The `id` property in the response serves as the unique ID for the key, which you can specify when invoking other CRUD operations.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:46]:
```suggestion
The response is a [Key object](#key-object) that represents the key that you create. The `id` property in the response serves as the unique ID for the key, which you can specify when invoking other CRUD operations.
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 8]
```diff
@@ -0,0 +1,360 @@
...
+The `keyId` provided in the response is the alias of the public key that can be used to get details of the public key data in a separate call.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:47]:
```suggestion
The `keyId` provided in the response is the alias of the public key that you can use to get details of the public key data in a separate call.
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 9]
```diff
@@ -0,0 +1,360 @@
...
+
+| Parameter | Description               | Param Type   | DataType   | Required |
+| --------- | ------------------------- | ------------ | ---------- | -------- |
+| `id`      | A valid key ID   | Path         | String     | TRUE     |
+
+> **Note:** The `?expand=publickey` query parameter optionally returns the full object including the details of the public key in the response body's `_embedded` property.
+
+##### Response parameters
+
+The response is a [key object](#key-object) that represents the key that matches the `id` you specify.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:98]:
```suggestion
The response is a [Key object](#key-object) that represents the key that matches the `id` that you specify.
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 10]
```diff
@@ -0,0 +1,360 @@
...
+  }
+}
+```
+
+> **Note:** The `keyId` is the alias of the public key that you can use to retrieve the public key.
+
+### Get a key
+
+<ApiOperation method="get" url="/api/v1/hook-keys/${id}" />
+
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:87]: Retrieves a specific key

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 11]
```diff
@@ -0,0 +1,360 @@
...
+  "lastUpdated": "2022-08-31T18:09:58.000Z",
+  "isUsed": false
+}]
+```
+
+### Get a public key
+
+<ApiOperation method="get" url="/api/v1/hook-keys/${keyId}" />
+
+Returns the public portion of the key object using the `keyId` parameter.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:167]:
```suggestion
Retrieves the public portion of the Key object using the `keyId` parameter
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 12]
```diff
@@ -0,0 +1,360 @@
...
+
+| Parameter | Description               | Param Type   | DataType   | Required |
+| --------- | ------------------------- | ------------ | ---------- | -------- |
+| `keyId`      | A valid key ID   | Path         | String     | TRUE     |
+
+> **Note:** `keyId` is the alias of the public key.
+
+##### Response parameters
+
+The response represents the key that matches the `keyId` you specify.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:179]:
```suggestion
The response represents the key that matches the `keyId` that you specify.
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 13]
```diff
@@ -0,0 +1,360 @@
...
+
+```bash
+curl -v -X GET \
+-H "Authorization: SSWS ${api_token}" \
+"https://${yourOktaDomain}/api/v1/hook-keys/${KeyId}"
+```
+
+##### Response example
+
+The details of the response can be found in [public key details](#public-key-details).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:191]:
```suggestion
You can find the details of the response in [public key details](#public-key-details).
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 14]
```diff
@@ -0,0 +1,360 @@
...
+  "use": null,
+  "e": "AQAB",
+  "n": "togCavBd7yQ7fuTEdvRURfK04LSgxqbKlptc7Gy9l6Swk2nhJbfIuB2Eezrn_JX-Azh-8usMMKau8vkq6AR2pW5mgDNIJG0OnCdwK24BdohxWlVQRZ5fD5DYg-noWwIbDjCGxMq3BKhOJP8cGAXx-olfrkC2NRy-HiJZOPiONII_GnlBiVS29BmULbbhKNeSGtoe90tR_OsXozfBsRRKFnF4ewQG3QwE23q6W3uNsUTr87fD9OO0FxNXyjnR3UrMTq36W6QLhqgLxlSiqxXGALnhK5z7WfXxyGfYrytrg2tFIhiHi0hT300xRI4A6SqhPFSCegx_O8iZQlTeaiHgjQ"
+}
+```
+
+### Update a key
+
+<ApiOperation method="put" url="/api/v1/hook-keys/${id}" />
+
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:207]: Replaces a key

I'm following Royal's guidelines for Redocly wording going forward

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 15]
```diff
@@ -0,0 +1,360 @@
...
+    "n": "2naqCnv6r4xNQs7207lRtKQvdtnlVND-8k5iYBIiqoKGY3CqUmRm1jleoOniiQoMkFX8Wj2DmVqr002efF3vOQ7_gjtTatBTVUNbNIQLybun4dkVoUtfP7pRc5SLpcP3eGPRVar734ZrpQXzmCEdpqBt3jrVjwYjNE5DqOjbYXFJtMsy8CWE9LRJ3kyHEoHPzo22dG_vMrXH0_sAQoCk_4TgNCbvyzVmGVYXI_BkUnp0hv2pR4bQVRYzGB9dKJdctOh8zULqc_EJ8tiYsS05YnF7whrWEyARK0rH-e4d4W-OmBTga_zhY4kJ4NsoQ4PyvcatZkxjPO92QHQOFDnf3w"
+  }
+}
+```
+
+### List keys
+
+<ApiOperation method="get" url="/api/v1/hook-keys" />
+
+Returns a list of keys
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:133]:
```suggestion
Lists all keys
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 16]
```diff
@@ -0,0 +1,360 @@
...
+##### Request parameters
+
+| Parameter  | Description                                                                     | Param Type   | DataType                                    | Required |
+| ---------- | ------------------------------------------------------------------------------- | ------------ | ------------------------------------------- | -------- |
+| id         | The ID of the key that you want to update                                   | Path         | String                                      | TRUE     |
+| key request | An object that represents the updated properties that you want to apply   | Body         |[key request object](#key-request-object)  | TRUE     |
+
+The submitted key request replaces the existing properties after passing validation.
+
+> **Note:** The only parameter that can be updated is the name of the key, which needs to be unique at all times.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:217]:
```suggestion
> **Note:** The only parameter that you can update is the name of the key, which needs to be unique at all times.
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 17]
```diff
@@ -0,0 +1,360 @@
...
+
+##### Response parameters
+
+The response is a [key object](#key-object) that represents the updated key.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:221]:
```suggestion
The response is a [Key object](#key-object) that represents the updated key.
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 18]
```diff
@@ -0,0 +1,360 @@
...
+    "e": "AQAB",
+    "n": "2naqCnv6r4xNQs7207lRtKQvdtnlVND-8k5iYBIiqoKGY3CqUmRm1jleoOniiQoMkFX8Wj2DmVqr002efF3vOQ7_gjtTatBTVUNbNIQLybun4dkVoUtfP7pRc5SLpcP3eGPRVar734ZrpQXzmCEdpqBt3jrVjwYjNE5DqOjbYXFJtMsy8CWE9LRJ3kyHEoHPzo22dG_vMrXH0_sAQoCk_4TgNCbvyzVmGVYXI_BkUnp0hv2pR4bQVRYzGB9dKJdctOh8zULqc_EJ8tiYsS05YnF7whrWEyARK0rH-e4d4W-OmBTga_zhY4kJ4NsoQ4PyvcatZkxjPO92QHQOFDnf3w"
+  }
+}
+```
+
+### Delete a key
+
+<ApiOperation method="delete" url="/api/v1/hook-keys/${id}" />
+
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:259]: Deletes a key

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 19]
```diff
@@ -0,0 +1,360 @@
...
+##### Request parameters
+
+| Parameter | Description                            | Param Type   | DataType   | Required |
+| --------- | -------------------------------------- | ------------ | ---------- | -------- |
+| `id`      | The ID of the key to delete   | Path         | String     | TRUE     |
+
+Deletes the key that matches the provided `id`. After it is deleted, the key is unrecoverable. As a safety precaution, only keys that are not being used are eligible for deletion.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:266]:
```suggestion
Deletes the key that matches the provided `id`. After it's deleted, the key is unrecoverable. As a safety precaution, only keys that aren't being used are eligible for deletion.
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 20]
```diff
@@ -0,0 +1,360 @@
...
+curl -v -X DELETE \
+-H "Authorization: SSWS ${api_token}" \
+"https://${yourOktaDomain}/api/v1/hook-keys/${id}"
+```
+
+##### Response example
+
+204 with no content
+
+## Key management objects
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:284]:
```suggestion
## Key Management objects
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 21]
```diff
@@ -0,0 +1,360 @@
...
+
+The key management API has the following objects:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:286]:
```suggestion
The Key Management API has the following objects:
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 22]
```diff
@@ -0,0 +1,360 @@
...
+
+* [Key request object](/#key-request-object)
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:288]:
```suggestion
* [Key Request object](/#key-request-object)
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 23]
```diff
@@ -0,0 +1,360 @@
...
+* [Key object](#key-object)
+* [Public key details](#public-key-details)
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:290]: Public Key Details

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 24]
```diff
@@ -0,0 +1,360 @@
...
+
+### Key request object
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:292]:
```suggestion
### Key Request object
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 25]
```diff
@@ -0,0 +1,360 @@
...
+
+| Property       | Description                                                                                         | DataType                            | Nullable   | Unique   | ReadOnly   | Validation                                        |
+| -------------- | --------------------------------------------------------------------------------------------------- | ----------------------------------- | ---------- | -------- | ---------- | ------------------------------------------------- |
+| name           | Display name for the key                                                                | String                              | FALSE      | TRUE     | FALSE      | Must be between 1 and 255 characters in length   |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:296]:
```suggestion
| name           | Display name for the key                                                                | String                              | FALSE      | TRUE     | FALSE      | Must be between one and 255 characters in length   |
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 26]
```diff
@@ -0,0 +1,360 @@
...
+}
+```
+
+### Key object
+
+| Property       | Description                                                                                         | DataType                            | Nullable   | Unique   | ReadOnly   | Validation                                        |
+| -------------- | --------------------------------------------------------------------------------------------------- | ----------------------------------- | ---------- | -------- | ---------- | ------------------------------------------------- |
+| id             | Unique key                                                                  | String                              | FALSE      | TRUE     | TRUE       | System assigned                                          |
+| keyId         | Alias of the public key                                      | String                              | FALSE      | TRUE    | TRUE      | System assigned            |
+| name           | Display name for the key                                                                   | String                              | FALSE      | TRUE     | FALSE      | Must be between 1 and 255 characters in length   |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:311]:
```suggestion
| name           | Display name for the key                                                                   | String                              | FALSE      | TRUE     | FALSE      | Must be between one and 255 characters in length   |
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 27]
```diff
@@ -0,0 +1,360 @@
...
+| isUsed           | Declares if this item is currently in use by other applications.   | String(Boolean)                      | FALSE      | FALSE    | TRUE       | System assigned             |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:312]:
```suggestion
| isUsed           | Declares if this item is currently in use by other applications   | String(Boolean)                      | FALSE      | FALSE    | TRUE       | System assigned             |
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 28]
```diff
@@ -0,0 +1,360 @@
...
+| created        | Date of key creation                                                                       | String (Date)                       | TRUE       | FALSE    | TRUE       | System assigned                                          |
+| lastUpdated    | Date of key update                                                                         | String (Date)                       | TRUE       | FALSE    | TRUE       | System assigned                                          |
+| \_embedded    | [Public key details](#public-key-details)                                                                         | JSON                     | FALSE     | TRUE   | TRUE      | System assigned                                        |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:315]:
```suggestion
| \_embedded    | [Public Key Details](#public-key-details)                                                                         | JSON                     | FALSE     | TRUE   | TRUE      | System assigned                                        |
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 29]
```diff
@@ -0,0 +1,360 @@
...
+    "alg": "RSA",
+    "kid": "7fbc27fd-e3df-4522-86bf-1930110256ad",
+    "use": null,
+    "e": "AQAB",
+    "n": "2naqCnv6r4xNQs7207lRtKQvdtnlVND-8k5iYBIiqoKGY3CqUmRm1jleoOniiQoMkFX8Wj2DmVqr002efF3vOQ7_gjtTatBTVUNbNIQLybun4dkVoUtfP7pRc5SLpcP3eGPRVar734ZrpQXzmCEdpqBt3jrVjwYjNE5DqOjbYXFJtMsy8CWE9LRJ3kyHEoHPzo22dG_vMrXH0_sAQoCk_4TgNCbvyzVmGVYXI_BkUnp0hv2pR4bQVRYzGB9dKJdctOh8zULqc_EJ8tiYsS05YnF7whrWEyARK0rH-e4d4W-OmBTga_zhY4kJ4NsoQ4PyvcatZkxjPO92QHQOFDnf3w"
+  }
+}
+```
+
+### Public key details
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:336]:
```suggestion
### Public Key Details
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 30]
```diff
@@ -0,0 +1,360 @@
...
+
+The public key details are defined in the `_embedded` property of the key object.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:338]:
```suggestion
The Public Key Details are defined in the `_embedded` property of the Key object.
```

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md [hunk 31]
```diff
@@ -0,0 +1,360 @@
...
+| -------------- | --------------------------------------------------------------------------------------------------- | ----------------------------------- | ---------- | -------- | ---------- | ------------------------------------------------- |
+ kty            | Cryptographic algorithm family for the certificate's keypair                                                                  | String                              | FALSE      | FALSE     | TRUE       | System assigned                                          |
+| alg         | Algorithm used in the key                                      | String                              | FALSE      | FALSE    | TRUE      | System assigned            |
+| kid           | Unique identifier for the certificate                                                                   | String                              | FALSE      | TRUE     | TRUE      | System assigned    |
+| use           |Acceptable usage of the certificate  | String                     | TRUE      | FALSE    | TRUE       | System assigned             |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:345]:
```suggestion
| use           |Acceptable use of the certificate  | String                     | TRUE      | FALSE    | TRUE       | System assigned             |
```

**Review by susanharper-okta — APPROVED** [review]
style things - approving to not hold you up.

File: packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md (hunk 6 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/reference/api/hook-keys/index.md:31]: I believe they want to expand the use of this API past hooks so were keeping this title generic

**Review by brianduffield-okta — APPROVED** [review]

**Pull request merged**